### PR TITLE
Fix bugs where tasklist was not being saved

### PR DESCRIPTION
**Pull request by LeeMingDe**

### CODE REVIEW

File: src/main/java/seedu/address/model/room/Room.java [hunk 1]
```diff
@@ -40,11 +40,13 @@ public Room(int roomNumber, boolean isOccupied) {
...
      * @param roomNumber Room Number of the room.
      * @param patient Patient to be added to the room.
+     * @param taskList TaskList of tasks for the room.
      */
-    public Room(int roomNumber, Patient patient) {
+    public Room(int roomNumber, Patient patient, TaskList taskList) {
         this.roomNumber = roomNumber;
         this.isOccupied = true;
         this.patient = patient;
+        this.taskList = taskList;
```
Comment (itssodium) [inline src/main/java/seedu/address/model/room/Room.java:49]: Do you want to have this as a separate constructor instead. So you would have Room(int roomNumber, Patient patient) and Room(int roomNumber, Patient patient, TaskList taskList).

File: src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java [hunk 2]
```diff
@@ -215,16 +225,18 @@ public boolean equals(Object other) {
...
         public String toString() {
             return "EditRoomDescriptor{"
-                + "roomNumber=" + roomNumber
-                + ", isOccupied=" + isOccupied
-                + ", patientName=" + patientName
-                + '}';
+                    + "roomNumber=" + roomNumber
+                    + ", isOccupied=" + isOccupied
+                    + ", patientName=" + patientName
+                    + ", taskList=" + taskList
```
Comment (itssodium) [inline src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java]: Good for changing the toString method accordingly.

File: src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java [hunk 3]
```diff
@@ -215,16 +225,18 @@ public boolean equals(Object other) {
             EditRoomDescriptor e = (EditRoomDescriptor) other; // state check
             return getRoomNumber().equals(e.getRoomNumber())
                     && getIsOccupied().equals(e.getIsOccupied())
-                    && getPatientName().equals(e.getPatientName());
+                    && getPatientName().equals(e.getPatientName())
+                    && getTaskList().equals(e.getTaskList());
```
Comment (itssodium) [inline src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java]: Good for changing the equals method.

**Review by itssodium — APPROVED** [review]
Good job! LGTM!

File: src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java [hunk 4]
```diff
@@ -95,15 +96,16 @@ private static Room createEditedRoom(Model model, Room roomToEdit,
                                          EditRoomDescriptor editRoomDescriptor) throws CommandException {
         assert (roomToEdit != null);
         int updatedRoomNumber = editRoomDescriptor.getRoomNumber().orElse(roomToEdit.getRoomNumber());
+        TaskList updatedTaskList = editRoomDescriptor.getTaskList().orElse(roomToEdit.getTaskList());
```
Comment (w-yeehong) [inline src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java]: I think `editroom` command does not allow task to be edited right?

In that case, maybe we don't need to use `editRoomDescriptor.getTaskList()`?

**Review by w-yeehong — APPROVED** [review]
LGTM! Thanks for addressing the bug.

File: src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java (hunk 4 above)
Comment (LeeMingDe) [reply src/main/java/seedu/address/logic/commands/room/EditRoomCommand.java]: yup, good catch. i removed it

**Pull request merged**